docs/Brew-Livecheck.md

```diff
@@ -96,6 +96,18 @@ end
 
 If tags include the software name as a prefix (e.g. `example-1.2.3`), it's easy to modify the regex accordingly: `/^example[._-]v?(\d+(?:\.\d+)+)$/i`
 
+### Referenced formula/cask
+
+A formula/cask can use the same check as another by using `formula` or `cask`.
+
+```ruby
+livecheck do
+  formula "another-formula"
+end
+```
+
+The referenced formula/cask should be in the same tap, as a reference to a formula/cask from another tap will generate an error if the user doesn't already have it tapped.
```

Comment:
Think an audit for this is needed or is just knowing that it will error on CI good enough? Think I'm fine either way

Reply:
I tried to create an audit for it but I'm not sure we can load a formula/cask in the context of a Rubocop. I don't have much experience with creating audits but my in-progress audit encounters an ```undefined method `metadata' for nil:NilClass``` error when run on a formula using `formula`/`cask` in a `livecheck` block:

```ruby
# This cop ensures that the argument to `#formula` or `#cask` only
# references a formula/cask in the same tap.
#
# @api private
class LivecheckReferenceSameTap < FormulaCop
  def audit_formula(_node, _class_node, _parent_class_node, body_node)
    tap_name = formula_tap
    return unless tap_name == "homebrew-core"

    livecheck_node = find_block(body_node, :livecheck)
    return if livecheck_node.blank?

    formula_node = find_every_method_call_by_name(livecheck_node, :formula).first
    formula_node_string = find_strings(formula_node).first if formula_node
    livecheck_formula = string_content(formula_node_string) if formula_node_string

    cask_node = find_every_method_call_by_name(livecheck_node, :cask).first
    cask_node_string = find_strings(cask_node).first if cask_node
    livecheck_cask = string_content(cask_node_string) if cask_node_string

    # Load the referenced formula or cask
    formula_or_cask, method_name, plural_name = if livecheck_formula.present?
      [Formulary.factory(livecheck_formula), "formula", "formulae"]
    elsif livecheck_cask.present?
      [Cask::CaskLoader.load(livecheck_cask), "cask", "casks"]
    end
    return unless formula_or_cask

    # Follow the formula/cask reference
    referenced_formula_or_cask, livecheck_references =
      Homebrew::Livecheck.resolve_livecheck_reference(formula_or_cask)

    # Compare the tap of each referenced formula/cask to formula_tap
    livecheck_references.each do |ref_formula_or_cask|
      if ref_formula_or_cask&.tap&.full_name&.split("/")&.last != tap_name
        offending_node(livecheck_node)
        problem "`#{method_name}` should only reference #{plural_name} in #{tap_name}."
      end
    end
  end
end
```

It would be nice to get this working but I'm stuck here at the moment and haven't dug into it further. In the interim time, I updated the other livecheck audits with respect to the new `formula`/`cask` methods.

---

homebrew/core CI's livecheck test would fail on a reference to a different tap, so that may be fine for now.

homebrew/cask CI taps both cask and core, so a reference to a homebrew/core formula in a tap wouldn't fail in that context. However, the audits we have for `livecheck` blocks don't apply to casks yet anyway (they're `FormulaCop`s), so this wouldn't change the status quo. I haven't looked into how we may be able to share audits for something that's used in both formulae/casks (e.g., `livecheck` blocks) but it's on my to-do list (along with bringing `livecheck` blocks in casks up to standard).